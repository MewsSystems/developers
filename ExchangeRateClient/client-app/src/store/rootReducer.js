// @flow

import {
  CONFIG_FETCH_SUCCESS,
  CONFIGS_FETCH_START,
  CONFIGS_FETCH_FAIL,
  RATES_FETCH_START,
  RATES_FETCH_SUCCESS,
  RATES_FETCH_FAIL,
  SELECT_RATES_IDS,
  type Action,
} from "../actions/configActions";
import type { StateTypes } from "./types";

export const initalState: StateTypes = {
  selectedRates: [],
  data: {},
  isFetchingConfig: false,
  APIerror: null,
  APIerror: null,
  isFetchingRates: false,
  rates: {},
};

export default (state: StateTypes = { ...initalState }, action: Action) => {
  switch (action.type) {
    case CONFIG_FETCH_SUCCESS:
      return {
        ...state,
        isFetchingConfig: false,
        ...action.payload,
      };
    case CONFIGS_FETCH_START:
      return {
        ...state,
        isFetchingConfig: true,
      };
    case CONFIGS_FETCH_FAIL:
      return {
        ...initalState,
        APIerror: action.payload.error,
      };
    case RATES_FETCH_START:
      return {
        ...state,
        selectedRates: action.payload,
        isFetchingRates: true,
      };
    case RATES_FETCH_SUCCESS: {
      const rates = action.payload;
      const ids = Object.keys(rates);

      const temp = ids.reduce(
        (acc, id) => ({
          ...acc,
          [id]: {
            current: rates[id],
            before: state.rates && state.rates[id] && state.rates[id].current,
          },
        }),
        {},
      );

      return {
        ...state,
        rates: temp,
        isFetchingRates: false,
      };
    }
    case RATES_FETCH_FAIL:
      return {
        ...state,
        isFetchingRates: false,
        APIerror: action.payload.error,
      };

    case SELECT_RATES_IDS:
      return {
        ...state,
        selectedRates: action.payload.ids,
      };

    default:
      return state;
  }
};
