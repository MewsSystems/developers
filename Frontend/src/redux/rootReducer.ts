import { combineReducers } from 'redux';
import searchReducer, { NAME as searchKey } from './searchReducer';
import movieReducer, { NAME as movieKey } from './movieReducer';
import configurationReducer, {
  NAME as configKey,
} from './configurationReducer';

const rootReducer = combineReducers({
  [searchKey]: searchReducer,
  [movieKey]: movieReducer,
  [configKey]: configurationReducer,
});

export default rootReducer;
