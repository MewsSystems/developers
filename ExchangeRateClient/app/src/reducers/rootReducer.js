import {combineReducers} from 'redux';
import config from './configReducer';

const rootReducer = combineReducers({
  config
});

export default rootReducer;