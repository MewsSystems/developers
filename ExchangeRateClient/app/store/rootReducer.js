import { combineReducers } from 'redux';
import {
    configReducer as config,
    currencyPairsReducer as currencyPairs,
    errorMessagesReducer as errorMessages,
    isFetchingReducer as isFetching,
    ratesHistoryReducers as ratesHistory,
    uiControlReducer as uiControl,
} from '../reducers';

const rootReducer = combineReducers({
    config,
    currencyPairs,
    errorMessages,
    isFetching,
    ratesHistory,
    uiControl,
});

export default rootReducer;
