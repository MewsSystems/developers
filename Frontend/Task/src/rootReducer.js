import { combineReducers } from 'redux';
import rates from './features/Rates/reducer';

const rootReducer = combineReducers({
  rates,
});

export default rootReducer;
