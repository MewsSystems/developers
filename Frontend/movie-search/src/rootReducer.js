import { combineReducers } from 'redux';
import { connectRouter } from 'connected-react-router';
import moviesReducer from './movies/MoviesReducer';

const createRootReducer = (history) => combineReducers({
  router: connectRouter(history),
  moviesReducer: moviesReducer
});

export default createRootReducer;
