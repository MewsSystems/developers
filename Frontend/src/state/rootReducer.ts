import {
  RecentMoviesState,
  recentMoviesInitialState,
  recentMoviesReducer,
} from 'state/reducers/movies/recent'
import { combineReducers } from 'redux'
import { PersistState } from 'redux-persist'

export interface State {
  recentMovies: RecentMoviesState
  _persist?: PersistState
}

export const initialState: State = {
  recentMovies: recentMoviesInitialState,
}

export const rootReducer = combineReducers({
  recentMovies: recentMoviesReducer,
})
