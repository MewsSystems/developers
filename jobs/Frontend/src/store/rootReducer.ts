/* Instruments */
import { TMDBApiSlice, SearchSlice } from '@/store/slices';

export const reducer = {
  [TMDBApiSlice.reducerPath]: TMDBApiSlice.reducer,
  [SearchSlice.reducerPath]: SearchSlice.reducer
};
