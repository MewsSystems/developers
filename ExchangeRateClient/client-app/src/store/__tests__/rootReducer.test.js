// @flow
import rootReducer, { initalState } from "../rootReducer";

describe("rootReducer", () => {
  test("should return initialState", () => {
    // $FlowTest
    const state = rootReducer(undefined, {});
    expect(state).toEqual(initalState);
  });

  test("should handle CONFIGS_FETCH_START action", () => {
    const state = rootReducer(initalState, { type: "CONFIGS_FETCH_START" });
    expect(state.isFetchingConfig).toEqual(true);
  });

  test("should handle CONFIG_FETCH_SUCCESS action", () => {
    const state = rootReducer(initalState, {
      type: "CONFIG_FETCH_SUCCESS",
      payload: {
        selectedRates: [
          "1993f7b9-f9be-551a-beac-312d6befd0cf",
          "41cae0fd-b74d-5304-a45c-ba000471eabd",
        ],
        data: {
          "41cae0fd-b74d-5304-a45c-ba000471eabd": {
            currencies: [
              { code: "PHP", name: "Philippines Peso" },
              { code: "DZD", name: "Algeria Dinar" },
            ],
          },
          "1993f7b9-f9be-551a-beac-312d6befd0cf": {
            currencies: [
              { code: "CZK", name: "Czech Republic Koruna" },
              { code: "LSL", name: "Lesotho Loti" },
            ],
          },
        },
      },
    });
    expect(state).toMatchSnapshot();
  });

  test("should handle RATES_FETCH_START action", () => {
    const state = rootReducer(initalState, {
      type: "RATES_FETCH_START",
      payload: ["1993f7b9-f9be-551a-beac-312d6befd0cf", "41cae0fd-b74d-5304-a45c-ba000471eabd"],
    });
    expect(state).toMatchSnapshot();
  });

  test("should handle RATES_FETCH_SUCCESS action", () => {
    const state = rootReducer(initalState, {
      type: "RATES_FETCH_SUCCESS",
      payload: {
        "1993f7b9-f9be-551a-beac-312d6befd0cf": 2.4015,
        "41cae0fd-b74d-5304-a45c-ba000471eabd": 2.4907,
      },
    });
    expect(state).toMatchSnapshot();
  });

  test("should handle CONFIGS_FETCH_FAIL action", () => {
    const state = rootReducer(initalState, {
      type: "CONFIGS_FETCH_FAIL",
      payload: {
        error: {
          config: {
            transformRequest: {},
            transformResponse: {},
            timeout: 240000,
            xsrfCookieName: "XSRF-TOKEN",
            xsrfHeaderName: "X-XSRF-TOKEN",
            maxContentLength: -1,
            headers: { Accept: "application/json, text/plain, */*" },
            method: "get",
            baseURL: "http://localhost:3000/",
            url: "http://localhost:3000/configuration",
            "axios-retry": { retryCount: 0, lastRequestTime: 1537218227843 },
          },
          request: {},
        },
      },
    });
    expect(state).toMatchSnapshot();
  });

  test("should handle RATES_FETCH_FAIL action", () => {
    const state = rootReducer(initalState, {
      type: "RATES_FETCH_FAIL",
      payload: {
        error: {
          config: {
            transformRequest: {},
            transformResponse: {},
            timeout: 240000,
            xsrfCookieName: "XSRF-TOKEN",
            xsrfHeaderName: "X-XSRF-TOKEN",
            maxContentLength: -1,
            headers: { Accept: "application/json, text/plain, */*" },
            method: "get",
            baseURL: "http://localhost:3000/",
            url: "http://localhost:3000/rates",
            "axios-retry": { retryCount: 0, lastRequestTime: 1537218227843 },
          },
          request: {},
        },
      },
    });
    expect(state).toMatchSnapshot();
  });

  test("should handle SELECT_RATES_IDS action", () => {
    const state = rootReducer(initalState, {
      type: "SELECT_RATES_IDS",
      payload: {
        ids: ["41cae0fd-b74d-5304-a45c-ba000471eabd", "5b428ac9-ec57-513d-8a08-20199469fb4d"],
      },
    });
    expect(state).toMatchSnapshot();
  });
});
