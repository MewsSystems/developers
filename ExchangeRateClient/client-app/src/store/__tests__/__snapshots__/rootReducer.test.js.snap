// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rootReducer should handle CONFIG_FETCH_SUCCESS action 1`] = `
Object {
  "APIerror": null,
  "data": Object {
    "1993f7b9-f9be-551a-beac-312d6befd0cf": Object {
      "currencies": Array [
        Object {
          "code": "CZK",
          "name": "Czech Republic Koruna",
        },
        Object {
          "code": "LSL",
          "name": "Lesotho Loti",
        },
      ],
    },
    "41cae0fd-b74d-5304-a45c-ba000471eabd": Object {
      "currencies": Array [
        Object {
          "code": "PHP",
          "name": "Philippines Peso",
        },
        Object {
          "code": "DZD",
          "name": "Algeria Dinar",
        },
      ],
    },
  },
  "isFetchingConfig": false,
  "isFetchingRates": false,
  "rates": Object {},
  "selectedRates": Array [
    "1993f7b9-f9be-551a-beac-312d6befd0cf",
    "41cae0fd-b74d-5304-a45c-ba000471eabd",
  ],
}
`;

exports[`rootReducer should handle CONFIGS_FETCH_FAIL action 1`] = `
Object {
  "APIerror": Object {
    "config": Object {
      "axios-retry": Object {
        "lastRequestTime": 1537218227843,
        "retryCount": 0,
      },
      "baseURL": "http://localhost:3000/",
      "headers": Object {
        "Accept": "application/json, text/plain, */*",
      },
      "maxContentLength": -1,
      "method": "get",
      "timeout": 240000,
      "transformRequest": Object {},
      "transformResponse": Object {},
      "url": "http://localhost:3000/configuration",
      "xsrfCookieName": "XSRF-TOKEN",
      "xsrfHeaderName": "X-XSRF-TOKEN",
    },
    "request": Object {},
  },
  "data": Object {},
  "isFetchingConfig": false,
  "isFetchingRates": false,
  "rates": Object {},
  "selectedRates": Array [],
}
`;

exports[`rootReducer should handle RATES_FETCH_FAIL action 1`] = `
Object {
  "APIerror": Object {
    "config": Object {
      "axios-retry": Object {
        "lastRequestTime": 1537218227843,
        "retryCount": 0,
      },
      "baseURL": "http://localhost:3000/",
      "headers": Object {
        "Accept": "application/json, text/plain, */*",
      },
      "maxContentLength": -1,
      "method": "get",
      "timeout": 240000,
      "transformRequest": Object {},
      "transformResponse": Object {},
      "url": "http://localhost:3000/rates",
      "xsrfCookieName": "XSRF-TOKEN",
      "xsrfHeaderName": "X-XSRF-TOKEN",
    },
    "request": Object {},
  },
  "data": Object {},
  "isFetchingConfig": false,
  "isFetchingRates": false,
  "rates": Object {},
  "selectedRates": Array [],
}
`;

exports[`rootReducer should handle RATES_FETCH_START action 1`] = `
Object {
  "APIerror": null,
  "data": Object {},
  "isFetchingConfig": false,
  "isFetchingRates": true,
  "rates": Object {},
  "selectedRates": Array [
    "1993f7b9-f9be-551a-beac-312d6befd0cf",
    "41cae0fd-b74d-5304-a45c-ba000471eabd",
  ],
}
`;

exports[`rootReducer should handle RATES_FETCH_SUCCESS action 1`] = `
Object {
  "APIerror": null,
  "data": Object {},
  "isFetchingConfig": false,
  "isFetchingRates": false,
  "rates": Object {
    "1993f7b9-f9be-551a-beac-312d6befd0cf": Object {
      "before": undefined,
      "current": 2.4015,
    },
    "41cae0fd-b74d-5304-a45c-ba000471eabd": Object {
      "before": undefined,
      "current": 2.4907,
    },
  },
  "selectedRates": Array [],
}
`;

exports[`rootReducer should handle SELECT_RATES_IDS action 1`] = `
Object {
  "APIerror": null,
  "data": Object {},
  "isFetchingConfig": false,
  "isFetchingRates": false,
  "rates": Object {},
  "selectedRates": Array [
    "41cae0fd-b74d-5304-a45c-ba000471eabd",
    "5b428ac9-ec57-513d-8a08-20199469fb4d",
  ],
}
`;
